File: IOT/Anshul-13J_AnshulNigam_2024cs1069_2/FunctionsAndArrays/AnyBaseToAnyBase.cpp
#include <iostream>
#include <math.h>
using namespace std;
int anyBaseToDecimal(int n,int base)
{
    int i=0,num=0,arr[15];
    while(n)
    {
        arr[i++]=n%10;
        n/=10;
    }
    for(int j=0;j<i;j++)
    {
        num += arr[j]*pow(base,j);
    }
   // cout<<num<<endl;
    return num;
    
}

int decimalToAnyBase(int n,int base)
{
    int i=0,num=0,arr[15];
    while(n)
    {
        arr[i++]= n%base ;
        n/=base;
    }
    for(int j=i-1;j>=0;j--)
        num=num*10 + arr[j];
    return num;
}
int main()
{
    int n,b1,b2,dec,num;
    cin>>n>>b1>>b2;
    dec= anyBaseToDecimal(n,b1);
    num= decimalToAnyBase(dec,b2);
    cout<<num;
    return 0;
}

File: Android/Tejash11_TejashSeth_2024csit1065_2/FunctionAndArrays/AnyBaseToAnyBase.java
import java.util.*;

public class Main {

  public static void main(String[] args) {
    Scanner scn = new Scanner(System.in);
    int n = scn.nextInt();
    int sourceBase = scn.nextInt();
    int  destBase = scn.nextInt();
    int value = getValue(n, sourceBase, destBase);
    System.out.println(value);
  }

  public static int getValue(int n, int src, int dest)
  {
    int decimalValue = anyBaseToDecimal(n, src);
    int finalBaseValue = decimalToAnyBase(decimalValue, dest);
    return finalBaseValue;
  }

  public static int anyBaseToDecimal(int n, int b)
  {
    int sum = 0, i = 0;
    while (n > 0)
    {
      sum += (n % 10) * Math.pow(b, i);
      i++;
      n = n / 10;
    }
    return sum;
  }

 
  public static int decimalToAnyBase(int n, int b)
  {
    int temp = 0;
    int r = 1;
    while (n > 0)
    {
      temp = temp + (n % b) * r;
      r *= 10;
      n = n / b;
    }
    return temp;
  }
}